internal/sql/pg/db.go

```diff
@@ -282,6 +289,28 @@ func (db *DB) BeginOuterTx(ctx context.Context) (sql.OuterTx, error) {
 	}, nil
 }
 
+// BeginChangesetTx starts a transaction that will return the entire changeset.
+// It works exactly like BeginOuterTx, but returns the full changeset body along
+// with the commit hash.
+func (db *DB) BeginChangesetTx(ctx context.Context, writer io.Writer) (*ChangesetTx, error) {
+	dbtx, err := db.BeginOuterTx(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	db.repl.changesetWriter.writer = writer
+	db.repl.changesetWriter.writable.Store(true)
```

Comment:
A related change should make a `Writer` associated to a specific transaction, avoiding issues if there are multiple write transactions (from any source whether another pg.DB instance or just another connection or store).  Probably extend `changesetIoWriter` to associate writers with the transaction sequence number (see `recvID` call in `precommit`).  That is, the `(rm *replMon) recvID` method would store the provided Writer in a map inside `changesetWriter` keyed by the same `seq int64` rather than just having a single writer that is used or not.

I'm fairly certain we have to do it this way otherwise any unsequenced write transactions, which are the ones that a `pg.DB` didn't make with an update to `kwild_internal.sentry`, will end up writing to the `Writer` incorrectly.

See the part in repl.go with `if seq == -1 {`.  While we can reset the hasher and not send the commit ID on the channel, we cannot unwrite whatever we gave to the Writer.

I can show you what I mean if this is fuzzy.  I have a feeling we will also have to move `incrementSeq` call so that it happens first, not as the last write of the outer txn; otherwise we can't identify at the start of the transaction's streaming updates if it's "our" transaction.